src/game_character.cpp

```diff
@@ -216,6 +216,9 @@ void Game_Character::UpdateMoveRoute(int32_t& current_index, const lcf::rpg::Mov
 	if (is_overwrite && !IsMoveRouteOverwritten()) {
 		return;
 	}
+	const auto num_commands = static_cast<int>(current_route.move_commands.size());
+	assert(current_index >= 0);
+	assert(current_index <= num_commands);
```

Comment:
Can this trigger when a hacked LDB is used? If yes this should be a warning and not an assert

Reply:
No, but it can trigger on a corrupted save game. I found this from an unrelated bug, that if the move route index is invalid it causes an infinite loop. The piece of code you previously commented on sanitizes the save data to prevent this from ever happening, and so the assert here is used to make sure that invariant is held.

I will add a comment here explaining this